src/main/java/com/fasterxml/jackson/core/filter/JsonPointerBasedFilter.java

```diff
@@ -13,25 +13,40 @@
 public class JsonPointerBasedFilter extends TokenFilter
 {
     protected final JsonPointer _pathToMatch;
+    protected final boolean _includeAllElements;
```

Comment:
Javadoc here (I realize there's lack of javadocs in this class prior to changes but.. hopefully can improve)... or
maybe the logic wrt "include all elements" needs to actually go in class Javadoc?